sherlockml/clients/auth.py

```diff
@@ -46,3 +47,54 @@ def get_access_token(self, client_id, client_secret):
         expires_at = now + timedelta(seconds=body['expires_in'])
 
         return AccessToken(token, expires_at)
+
+
+class SherlockMLAuth(AuthBase):
+    """Requests auth implementation for accessing SherlockML services.
+
+    Parameters
+    ----------
+    auth_service_url : str
+        The URL of the SherlockML authentication service
+    client_id : str
+        The SherlockML client ID to use for authentication
```

Comment:
I think due to the context we could drop `SherlockML` from this parameter description.

Reply:
Possibly, though I thought in this case being as unambiguous as possible might be advantageous.

I'm happy to drop it though if people feel that is better - I expect this class will not be used directly by casual users, but rather only by people developing clients in the library.